### DISCUSSION TITLE
[help wanted] Is new API only available with esm? (v2.0.0)

### DISCUSSION THREAD

**Discussion by developersoom** [pmndrs/jotai#1756] [Q&A]
```
Unhandled Runtime Error
ReferenceError: exports is not defined

../shared/dist/stores/file.js @ eval

   9 | }
  10 | });
> 11 | var _jotai = require("jotai");
     |         ^
  12 | var Atom = (0, _jotai.atom)({

```
This error pops up when I run my project locally. 
I've encountered this issue migrating to the new API (`v2.0.0`). Since `v1.9.2` worked well, I have no idea where to look at. 
My typescript version is `4.8.0.` (I've tried `4.7.0` but not worked either.) 
And this project is built with `next 12`. 
Is new API only worked with esm? or could you help me with this issue? 
Thanks!

**Comment by dai-shi** [comment]
Thanks for reporting.
https://github.com/pmndrs/jotai/pull/1675 should be related.

@barelyhuman What do you think?

**Reply by barelyhuman** [reply]
@developersoom Hey thanks for the versions, but if you could help with a mini reproduction for this we could help. 
For the tagged issue by @dai-shi , you could also try by readding the `module` key that was removed from the `exports` and see if that's fixing it for you. If not then the mini reproduction would help 😄

**Reply by developersoom** [reply]
Thank you for the help! Have tried to readd the `module` key, but still the same. Do I also need to readd **createESMConfig(`src/${c}.ts`, `dist/esm/${c}.js`)** in the `rollup.config.js`? Is it possible to change the `rollup.config.js` file since I cannot find it. Please let me know! Thank you.

**Comment by dai-shi** [comment]
@developersoom Hi, can you create a small reproduction? We would like to fix it.

**Reply by developersoom** [reply]
Thank you for trying to help. I'm having trouble with creating it with codesandbox. I'll post it as soon as it's done. Thanks!

**Reply by barelyhuman** [reply]
It most probably won’t break on Code sandbox, a local one with a git repo would be easier to break

**Comment by developersoom** [comment]
It seems to be related to `next-transpile-modules`. We've used  `next-transpile-modules 9.0.0` for old browsers. 

```
// next.config.js
const withTM = require('next-transpile-modules')(['jotai']);
module.exports = withTM({...})
```

Without it, it works fine! Thanks for the help.

**Reply by dai-shi** [reply]
@barelyhuman Do we think this as resolved? We could revert #1675, it it was too early to drop. Maybe it's just fine, as the entire ecosystem is moving forward.

**Reply by barelyhuman** [reply]
I'm not sure how `next-transpile-module` works so would want to check that before I make a statement

**Reply by barelyhuman** [reply]
I went through it and it's webpack's older issues with  resolution of multi-entry packages causing it. We could turn back based on how many of these issues pop up, I wouldn't revert on 1 issue since if we keep doing this, the pure ESM and simplified packages might be a hard goal to achieve altogether. 

Let's wait on how many are facing this

**Reply by dai-shi** [reply]
Yeah, I agree.

> Maybe it's just fine, as the entire ecosystem is moving forward.

Thanks for your investigation!
